### ISSUE TITLE
Redirect to login.html is incorrect when using a reverse proxy

### ISSUE THREAD

**Issue by sstallion** [AdguardTeam/AdGuardHome#6604]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

FreeBSD, AMD64 (aka x86_64)

### Installation

Custom package (OpenWrt, HomeAssistant, etc; please mention in the description)

### Setup

On one machine

### AdGuard Home version

0.107.36

### Action

When using AdGuard Home behind a reverse proxy (Apache 2.4 in this case), unauthenticated users are redirected to `/login.html`, which causes issues when placed behind a subdirectory. The naive fix for this would be a relative redirect (eg. `login.html`.

My Apache configuration (which should be equivalent to the documented nginx configuration) is:
```
ProxyPreserveHost On
ProxyPass /adguard/ http://localhost:3000/
ProxyPassReverse /adguard/ http://localhost:3000/
ProxyPassReverseCookiePath /adguard/ /
```

Thanks in advance!

### Expected result

Redirects work when using a reverse proxy from a subdirectory (eg. `/adguard/login.html`).

### Actual result

An absolute redirect to `/login.html` is performed.

### Additional information and/or screenshots

FreeBSD 13.2-RELEASE-p9; installed using the `www/adguardhome` port.

**Comment by overwatch3560** [comment]
@sstallion are you still experiencing this issue ?

**Comment by sstallion** [comment]
> @sstallion are you still experiencing this issue ?

I am.

**Comment by sstallion** [comment]
> @sstallion, hello and apologies for the late response. Unfortunately, we can't reproduce it with nginx, following [this guide](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#nginx). However, I think I've noticed a mistake in the shared Apache configuration – according to the [documentation](https://httpd.apache.org/docs/2.4/ru/mod/mod_proxy.html#proxypassreversecookiepath), the `ProxyPassReverseCookiePath` directive behaves just like [`proxy_cookie_path`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cookie_path) in nginx, looking for the first argument to rewrite it with the second, and not vice versa. Could you please try to run the proxy with the change?

Nice catch! I fixed the configuration, but unfortunately I'm seeing the same behavior.

**Comment by EugeneOne1** [comment]
@sstallion, hello and apologies for the late reply.  We've been trying different configurations of the Apache HTTP server and haven't really come up with a reliable one.  The thing is, we can't just use the relative URLs without leading slashes, because sometimes AdGuard Home redirects to the root (`http://localhost:3000/`) of the domain, which is not possible to perform relatively.

Changing the `Location` header is actually the work for the reverse HTTP proxy.  The `ProxyPassReverse` directive seems to be intended for this, but we can't get it to work with relative redirections (it worked with an absolute URL in the header, but AdGuard Home doesn't know the hostname of the actual URL).  AFAIK there might be some workarounds like using the `RewriteRule` directive, but in general the Apache reverse HTTP proxy is not recommended to be used behind AdGuard Home, please consider using the alternatives mentioned in the [knowledge base](https://adguard-dns.io/kb/adguard-home/faq/#reverseproxy).

**Comment by naanlizard** [comment]
Happens with nginx too

https://github.com/AdguardTeam/AdGuardHome/blob/bf1101b4603ced00e6333dd77c8a0f78a867f088/internal/home/authhttp.go#L279

This line always redirects to domain.tld/login.html

If your adguard install is in a subfolder (e.g. domain.tld/adguard/) it will redirect to the wrong place

```
location /adguard/ {

    include /config/nginx/proxy.conf; (from [docker-swag](https://github.com/linuxserver/docker-swag/blob/79f6dd4cb19ed2e186ddeb048350c6124e17219a/root/defaults/nginx/proxy.conf.sample))
    proxy_cookie_path / /adguard/;

    proxy_pass http://adguardIP:80/;

}
```

**Comment by binarrii** [comment]
I encountered this problem.

- Enable encryption, enable `DoT` and `DoQ`, but disable `HTTPS`.
- `DoT` and `DoQ` requests go directly to AdguardHome, while `HTTPS/DoH` requests reach AdguardHome through **Openresty/Nginx**.
- When saving the encrypted configuration, you will be redirected to `http://$host:3000`
- My configuration and code logic errors are shown in the pictures below.
- I configured it this way to better customize/hide the access path to the admin page, as I do not want the admin page to be directly accessible from the root path.
- Also see:    [https://github.com/AdguardTeam/AdGuardHome/issues/7003](https://github.com/AdguardTeam/AdGuardHome/issues/7003)

![Image](https://github.com/user-attachments/assets/c864eaf3-44ab-46ee-a820-23a78d6263cc)

![Image](https://github.com/user-attachments/assets/2fc78acd-51b2-489a-98b8-8fda2389a59c)

- Related code: `https://github.com/AdguardTeam/AdGuardHome/blob/master/client/src/helpers/helpers.tsx#L300`

![Image](https://github.com/user-attachments/assets/5b37e754-c6fa-43fb-abcc-528002bcc99e)

**Comment by cizordj** [comment]
I am using cloudflared as a reverse-proxy and a similar issue occurs when I access via the public hostname.

```txt
$ curl -v --compressed https://dns.example.com/ -L
> GET / HTTP/2
> Host: dns.example.com
> User-Agent: curl/8.21.0
> Accept: */*
> Accept-Encoding: deflate, gzip, br, zstd
>
* Request completely sent off
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
< HTTP/2 302
< date: Fri, 14 Aug 2026 22:59:17 GMT
< content-type: text/html; charset=utf-8
< location: http://
< server: cloudflare
< alt-svc: h3=":443"; ma=86400
* Ignoring the response-body

* Connection #0 to host dns.example.com:443 left intact
* The redirect target URL could not be parsed: No host part in the URL
curl: (3) The redirect target URL could not be parsed: No host part in the URL
```

It seems that AdGuard is returning an invalid URL and curl says it in plain English:
`The redirect target URL could not be parsed: No host part in the URL`

The http header confirms:
`< location: http://`

However, when I go to `dns.example.com/login.html` it works normally.

**Comment by droscy** [comment]
I've found a working solution for Apache using the [mod_headers](https://httpd.apache.org/docs/2.4/ru/mod/mod_headers.html) module to change the `Location` header

```
<Location /aghome/>
    ProxyPreserveHost On
    ProxyPass http://localhost:3000/
    ProxyPassReverse http://localhost:3000/
    Header edit Location ^/login.html$ /aghome/login.html
</Location>
```
